packages/flutter_markdown/lib/src/builder.dart

```diff
@@ -184,12 +274,83 @@ class MarkdownBuilder implements md.NodeVisitor {
       node.accept(this);
     }
 
+    // We loop through all widgets to filter out [Anchor] and return
+    // [IndexedAnchorData] so we know which widget index we need to scroll to
+    // to get to a certain anchor inside the rendered markdown text.
+    //
+    // We need the index for scrolling between items because we use
+    // [ScrollablePositionedList] (which requires an index for scrolling to a
+    // widget inside the list).
+    final List<Widget> widgets = _flattenWidgets(_blocks.single.children);
+    _anchorsWithIndex = _getAnchorsWithIndex(widgets);
+
     assert(_tables.isEmpty);
     assert(_inlines.isEmpty);
     assert(!_isInBlockquote);
     return _blocks.single.children;
   }
 
+  List<IndexedAnchorData> _getAnchorsWithIndex(List<Widget> widgets) {
+    final List<IndexedAnchorData> _anchors = <IndexedAnchorData>[];
+
+    for (int i = 0; i < widgets.length; i++) {
+      final Widget widget = widgets[i];
+      if (widget is Anchor) {
+        // If we have already an index for this anchor, i.e. multiple anchors
+        // with the same id (which *should* never happen) then we use the first
+        // occasion of the anchor.
+        if (getIndexForAnchor(widget.data.id) != null) {
+          continue;
+        }
+        _anchors.add(IndexedAnchorData(i, widget.data.id, widget.data.text));
+      }
+    }
+
+    return _anchors;
+  }
+
+  /// For all [widgets] we get their children and the children of these children
+  /// (etc) and return them all in a "flattened" list.
+  /// In this way we basically squish a whole widget "tree" into one layer so we
+  /// can later search for a specific widget ([Anchor]).
+  List<Widget> _flattenWidgets(List<Widget> widgets) {
+    final List<Widget> _widgets = <Widget>[];
+
+    _widgets.addAll(widgets);
+    for (final Widget widget in widgets) {
+      final List<Widget>? children = _tryGetChildrenCombined(widget);
+      if (children != null) {
+        _widgets.addAll(_flattenWidgets(children));
+      }
+    }
+    return _widgets;
+  }
+
```

Comment:
@domesticmouse Do you have any feedback regarding this? Is there a better way than to loop through all widgets and flattening them? Also is there a better way to flatten them instead of casting them to dynamic and accessing `.child` or `.children`? This feels very hacky.

Reply:
Flattening the tree feels wrong to me. What are you attempting to do here?

Reply:
> Flattening the tree feels wrong to me.

Yeah same :D

I want to know which index the `Anchor` widgets have inside the widget list that we return inside `MarkdownBuilder.build` so we can scroll to a specific anchor via `ScrollablePositionedList`.    

If I don't flatten the widget list then I only loop through the "top most" widgets (I think SizedBoxes and Columns or sth like this) and don't encounter any `Anchor` widgets. Thus I wouldn't know what indexes the `Anchor` widgets have and couldn't scroll to them later.   

I think there will probably a much better way to solve this - thats the reason why I opened this PR because I'm not sure how.

Reply:
@stuartmorgan any ideas on who to rope in here?